internal/collector/storageclass.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"k8s.io/kube-state-metrics/pkg/metric"
+
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descStorageClassLabelsName          = "kube_storageclass_labels"
+	descStorageClassLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descStorageClassLabelsDefaultLabels = []string{"storageclass"}
+
+	storageClassMetricFamilies = []metric.FamilyGenerator{
+		{
+			Name: "kube_storageclass_info",
+			Type: metric.Gauge,
+			Help: "Information about storageclass.",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				m := metric.Metric{
+					LabelKeys:   []string{"provisioner", "reclaimPolicy", "volumeBindingMode"},
+					LabelValues: []string{s.Provisioner, string(*s.ReclaimPolicy), string(*s.VolumeBindingMode)},
+					Value:       1,
+				}
+				return &metric.Family{Metrics: []*metric.Metric{&m}}
+			}),
+		},
+		{
+			Name: "kube_storageclass_created",
+			Type: metric.Gauge,
+			Help: "Unix creation timestamp",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				ms := []*metric.Metric{}
+
+				if !s.CreationTimestamp.IsZero() {
+					ms = append(ms, &metric.Metric{
+						Value: float64(s.CreationTimestamp.Unix()),
+					})
+				}
+
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_storageclass_metadata_resource_version",
+			Type: metric.Gauge,
+			Help: "Resource version representing a specific version of storageclass.",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   []string{"resource_version"},
+							LabelValues: []string{string(s.ObjectMeta.ResourceVersion)},
```

Comment:
As far as I know `ResourceVersion` is increased for each change propagated from etcd. Thus this metric would create a new timeseries for each change of the object. How often do you expect this Kubernetes object to change?

Reply:
The `ResourceVersion` can be changed only when object is updated, but we won't update it usually, so I think it will not change freequently.